File: 0202.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        # showed num
        arr = [n]
        
        while 1:
            digits = [int(i) for i in str(n)]
            n = 0
            for i in digits:
                n += i * i
            if n in arr:
                break
            arr.append(n)
        return (n == 1)

File: 0202.cpp
class Solution {
public:
    bool isHappy(int n) {
        vector<bool> seen(1000, false);

        while (n != 1) {
            int sum = 0;
        
            while (n > 0) {
                sum += (n % 10) * (n % 10);
                n /= 10;
            }
            n = sum;
            if (seen[n]) {
                return false;
            }
            seen[n] = true;
            
        }

        return true;
    }
};
